tests/manage/monitoring/conftest.py

```diff
@@ -62,6 +63,12 @@ def stop_mgr():
     measured_op = measure_operation(stop_mgr, test_file)
     logger.info(f"Upscaling deployment {mgr} back to 1")
     oc.exec_oc_cmd(f"scale --replicas=1 deployment/{mgr}")
+
+    # wait for ceph to return into HEALTH_OK state after mgr deployment
+    # is returned back to normal
+    ceph_cluster = CephCluster()
+    ceph_cluster.cluster_health_check()
```

Comment:
I must be missing something. How come this won't fail?

I see that the original problem was that after issuing a request to scale a ceph deployment back to the original number, it takes some time for the cluster to recover after the missing component is back. So while the cluster recovered, a check for ceph health in teardown failed because the recovery was not fast enough. So how come this healt check would not fail and instead wait for the cluster to sort it out? Would the default timeout in `cluster_health_check()` choice work every time this is executed?

That said, if you demonstrate that it works when executed multiple times in a PR validation job, I'm ok with this as it would be still improvement, preventing false failures.

Reply:
As described in docstring for `cluster_health_check` function, if no timeout parameter is provided then the timeout will be scaled based on number of ceph pods in the cluster (which is `10 * number of ceph pods`). This gives cluster enough time to recover.

Reply:
I'm bit reluctant to rely on a function which on one hand is called `cluster_health_check()`, but on the other hand is trying health check multiple times (based on some magic heuristic) and only fails if all these retries fail. Besides that, it also checks number of mons, mds and some noobaa stuff. The main functionality we need here is just a side effect of a design choice burried in this function (only when I actually check the source code I can understand why it could work, looking at the function name or a docstring makes me confused).

Have you tried to run it multiple times as I suggested? Have you tried to see what happens when this  `cluster_health_check()` function fails?

Reply:
I tried it 3x times. I didn't see it fail. Do you suggest to change the function in this PR or create a new `cluster_health_check` function?

Reply:
@fbalak how about waiting for Ceph health to be OK, with a fixed timeout, regardless of validating different resources like MDS and Noobaa? 
You can use ocs_ci.utility.utils.ceph_health_check

Reply:
Ok, ocs_ci.utility.utils.ceph_health_check looks better.

Reply:
I agree, usage of the other health check function is more explicit and should be easier to debug if something goes wrong.